src/session.rs

```diff
@@ -162,6 +164,71 @@ impl<'session, M: FnMut(InboundMessage) + Send, E: FnMut(SessionEvent) + Send>
         }
         Ok(())
     }
+
+    pub fn endpoint_provision(
+        &self,
+        endpoint_props: EndpointProps,
+        ignore_already_exists_error: bool,
+    ) -> Result<()> {
+        let mut flag = ffi::SOLCLIENT_PROVISION_FLAGS_WAITFORCONFIRM;
+        if ignore_already_exists_error {
+            flag |= ffi::SOLCLIENT_PROVISION_FLAGS_IGNORE_EXIST_ERRORS;
+        }
+
+        let rc = unsafe {
+            ffi::solClient_session_endpointProvision(
+                endpoint_props.to_raw().as_mut_ptr(),
```

Comment:
You need to keep a reference around for the `vec` returned by `to_raw` or it could not live long enough.

Reply:
Fixed with [267de7d](https://github.com/asimsedhain/solace-rs/pull/27/commits/267de7d40c1d43e8f485db1c79074ad1c5dc5e22)